### PR TITLE
[Azure.Monitor.Query]Update test-resources.bicep

### PR DESCRIPTION
**Pull request by nisha-bhatia**
Update Monitor Query tests to not use the Tables API (got hung)
Remove MetricsBatch for a minor version release

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.Query/tests/LogsTestData.cs [hunk 1]
```diff
@@ -111,26 +111,36 @@ private async Task InitializeData(string workspaceId, string workspaceKey)
...
+            while (!succeeded)
             {
                 await Task.Delay(TimeSpan.FromSeconds(30));
-                count = await QueryCount(workspaceId);
+                succeeded = await QueryCount(workspaceId);
             }
         }
 
-        private async Task<int> QueryCount(string workspaceId)
+        private async Task<bool> QueryCount(string workspaceId)
```
Comment (JoshLove-msft) [inline sdk/monitor/Azure.Monitor.Query/tests/LogsTestData.cs:121]: Should we override IsEnvironmentReady for this logic?
https://grep.app/search?q=isenvironmentready

File: sdk/monitor/Azure.Monitor.Query/tests/LogsTestData.cs [hunk 2]
```diff
@@ -111,26 +111,36 @@ private async Task InitializeData(string workspaceId, string workspaceKey)
...
                 var countResponse = await logsClient.QueryWorkspaceAsync<int>(workspaceId, $"{TableAName} | count", DataTimeRange);
                 var count = countResponse.Value.Single();
-                return count;
+                var customColumnsQuery = await logsClient.QueryWorkspaceAsync<int>(workspaceId, $"{TableAName}" +
+                    $" | distinct * |" +
+                    $"project {StringColumnName}, {IntColumnName}, {BoolColumnName}, {FloatColumnName} |" +
+                    $"count",
+                    DataTimeRange);
+                var customColumns = customColumnsQuery.Value.Single();
+                if (count > 0 && customColumns > 0)
```
Comment (JoshLove-msft) [inline sdk/monitor/Azure.Monitor.Query/tests/LogsTestData.cs:134]: Do we need both count and customColumnsCount?

**Review by JoshLove-msft — APPROVED** [review]

**Comment by TimothyMothra** [comment]
@nisha-bhatia @JoshLove-msft 
Can we revert THIS PR? This seems to have broken the Azure.Monitor.Query tests.
All PRs in the monitor directory are now blocked. #41820 #41767

```
 Failed CanQueryBatchWithTimespan [29 ms]
  Error Message:
   Azure.Core.TestFramework.TestRecordingMismatchException : If this is a new recording, make sure you have pushed the recordings to the assets repository. For instructions, see: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core.TestFramework/README.md#recording

Unable to find a record for the request POST https://api.loganalytics.io/v1/$batch
Header differences:
Body differences:
Request and record bodies do not match at index 160:
     request: ""2024-01-24T00:00:00.000Z\"), "
     record:  ""2024-01-23T00:00:00.000Z\"), "
```

**Pull request merged**